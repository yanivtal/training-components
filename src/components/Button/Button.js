/* @flow */
import React from 'react'
import Theme from 'js-theme'

type PropsT = {
  children: React.Children,
  theme: Object,
}

const Button = ({
  children,
  theme,
}: PropsT) => (
  <button
    {...theme.button}
  >
    {children}
  </button>
)

const defaultTheme = {
  button: {
    whiteSpace: 'nowrap',
    userSelect: 'none',
    textDecoration: 'none',
    boxSizing: 'border-box',
    cursor: 'pointer',
    border: 0,
    backgroundColor: '#efefef',
  },
}

export default Theme('Button', defaultTheme)(Button)
