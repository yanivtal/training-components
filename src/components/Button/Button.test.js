import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

it('Renders without props', () => {
  const button = renderer.create(
    <Button />
  ).toJSON()
  expect(button).toMatchSnapshot()
})
