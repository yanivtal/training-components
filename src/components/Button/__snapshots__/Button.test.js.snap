exports[`test Renders without props 1`] = `
<button>
  Button
</button>
`;
